rust/core-lib/src/view.rs

```diff
@@ -662,14 +674,17 @@ impl View {
             }
         }
 
-        let styles =
-            self.render_styles(client, styles, start_pos, pos, &selections, &hls, style_spans);
-
-        let mut result = json!({
-            "text": &l_str,
-            "styles": styles,
-        });
+        let mut result = json!({});
 
+        if let Some(text) = text {
+            let l_str = text.slice_to_cow(start_pos..pos);
+            result["text"] = json!(&l_str);
+        }
```

Comment:
Untested, but imho something like this would be nicer than having the temporary variable `l_str`:

```suggestion
        if let Some(text) = text.map(|s| s.slice_to_cow(start_pos..pos)) {
            result["text"] = json!(&text);
        }
```

Reply:
I'll inline it into `json!(...)`. Combining "if let" and map() can be harder for understanding.